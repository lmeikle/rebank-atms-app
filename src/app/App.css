.app {
  max-width: 400px;
  margin-left: auto;
  margin-right: auto;
  text-align: center;
  font-family: 'Titillium Web', sans-serif;
  color: #002742;
}

a{text-decoration:none}
a{color:#1e70bf}
a:hover{text-decoration:none}
a:hover{color:#ED3773}

.app-topbar {
  height: 70px;
  background-color: #FFFFFF;
}

.app-topbar-logo {
  margin-top: 20px;
  width: 40%;
}
