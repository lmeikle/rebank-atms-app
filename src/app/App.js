import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import AtmsContainer from '../atms/AtmsContainer';
import BanksContainer from '../banks/BanksContainer';
import logo from './logo.svg';
import './App.css';

class App extends Component {
	render()
	{
		return (
			<div className="app">
				<div className="app-topbar">
					<img src={logo} className="app-topbar-logo" alt="logo"/>
				</div>
				<Router>
					<Switch>
						<Route path="/atms" component={AtmsContainer}/>
						<Route path="/" component={BanksContainer}/>
					</Switch>
				</Router>
			</div>
		);
	}
}

export default App;
