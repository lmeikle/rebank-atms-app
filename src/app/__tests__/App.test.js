import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

const app = shallow(<App />);

test('renders correctly', () => {
  expect(app).toMatchSnapshot();
});
