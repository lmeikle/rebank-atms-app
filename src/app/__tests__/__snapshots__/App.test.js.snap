// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="app-topbar"
>
          <img
                    alt="logo"
                    className="app-topbar-logo"
                    src="logo.svg"
          />
</div>,
        <BrowserRouter>
          <Switch>
                    <Route
                              component={[Function]}
                              path="/atms"
                    />
                    <Route
                              component={[Function]}
                              path="/"
                    />
          </Switch>
</BrowserRouter>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="logo"
            className="app-topbar-logo"
            src="logo.svg"
/>,
          "className": "app-topbar",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "className": "app-topbar-logo",
            "src": "logo.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Switch>
            <Route
                        component={[Function]}
                        path="/atms"
            />
            <Route
                        component={[Function]}
                        path="/"
            />
</Switch>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/atms"
/>,
              <Route
                component={[Function]}
                path="/"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/atms",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="app-topbar"
>
            <img
                        alt="logo"
                        className="app-topbar-logo"
                        src="logo.svg"
            />
</div>,
          <BrowserRouter>
            <Switch>
                        <Route
                                    component={[Function]}
                                    path="/atms"
                        />
                        <Route
                                    component={[Function]}
                                    path="/"
                        />
            </Switch>
</BrowserRouter>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="logo"
              className="app-topbar-logo"
              src="logo.svg"
/>,
            "className": "app-topbar",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "app-topbar-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Switch>
              <Route
                            component={[Function]}
                            path="/atms"
              />
              <Route
                            component={[Function]}
                            path="/"
              />
</Switch>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  path="/atms"
/>,
                <Route
                  component={[Function]}
                  path="/"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/atms",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
